fasthttpproxy/http.go

```diff
@@ -86,6 +99,10 @@ func FasthttpHTTPDialerTimeout(proxy string, timeout time.Duration) fasthttp.Dia
 			conn.Close()
 			return nil, fmt.Errorf("could not connect to proxy: %s status code: %d", proxy, res.Header.StatusCode())
 		}
+		if err := unsetDeadline(conn); err != nil {
+			conn.Close()
+			return nil, err
+		}
```

Comment:
I think it's nice to also just do this in a `if timeout > 0 {` instead of the `unsetDeadline` construction you have now.

Reply:
https://github.com/valyala/fasthttp/blob/926da8124a40a39d2a810d6502a48bfbf7866798/fasthttpproxy/http.go#L67

```unsetDeadline``` will do nothing if timeout is zero.

Reply:
> Yeah but it's better to just add `if timeout > 0 {` around this instead of using that closure variable.

OK